docs/proposals/34-observability2.0-opentelemtry-operator-and-collectors.md

```diff
@@ -0,0 +1,153 @@
+- **Details:** For each shoot, deploy an OpenTelemetry collector instance in its control plane namespace (e.g., `shoot--project--name`). The collector will run as a `Deployment` managed by the OpenTelemetry operator.
+- Configuration: During the `Shoot` reconciliation flow, an [`OpenTelemetryCollector` Custom Resource](https://github.com/open-telemetry/opentelemetry-operator/blob/main/apis/v1beta1/opentelemetrycollector_types.go) will be created in the `Shoot` namespace. This Custom Resource will trigger the Operator to create an OpenTelemetry Collector Deployment & Ingress inside the namespace.
+- **Example OpenTelemetryCollector CR:**
+
+```yaml
+apiVersion: opentelemetry.io/v1beta1
+kind: OpenTelemetryCollector
+metadata:
+  labels:
+    resources.gardener.cloud/managed-by: gardener
+  name: opentelemetry-collector
+spec:
+  mode: deployment
+  config:
+    exporters:
+      loki:
+      endpoint: https://v-local--local.ingress.local.seed.local.gardener.cloud/vali/api/v1/push
+        headers:
+        # Authenticate against Vali using a Bearer Token
+        Authorization: "Bearer ${file:{{ .pathAuthToken }}}"
+    receivers:
+      otlp:
+        protocols:
+          http:
+            endpoint: 0.0.0.0:4317
+    service:
+      pipelines:
+        logs:
+          exporters:
+          - loki
+          receivers:
+          - otlp
+```
+
+![Figure 1](assets/gep-34-otel-operator-architecture.png)
+
+### 3. Integrate with Existing Components
+
+- **Objective:** Ensure a smooth transition by interfacing with the current observability stack.
+- **Details:**
+  - **Fluent-bit Integration:** Configure Fluent-bit to forward logs to the collector’s `loki` receiver, maintaining compatibility with the Vali-based setup from *GEP-19*.
+  - **Vali Integration:** Use the `loki` exporter to output the shoot logs to the existing Vali backend.
+  - **Valitail Replacement:** Replace the current `valitail` system service on shoot nodes with the OpenTelemetry collector, ensuring it collects the same types of logs as before. This will be done by deploying a systemd service for the collector on each shoot node, in the same way as the current `valitail` service.
+  - **Event Logger Replacement:** In a later stage the current `event logger` shall be replaced with the `k8s-event` receiver in  the otel-collector shoot instance
+
+![Figure 2](assets/gep-34-shoot-observability-architecture.png)
+
+Example Systemd unit for the OpenTelemetry Collector in the shoot node:
+
+```ini
+[Unit]
+Description=opentelemetry-collector daemon
+Documentation=https://github.com/open-telemetry/opentelemetry-collector
+[Install]
+WantedBy=multi-user.target
+[Service]
+CPUAccounting=yes
+MemoryAccounting=yes
+CPUQuota=3%
+CPUQuotaPeriodSec=1000ms
+MemoryMin=29M
+MemoryHigh=400M
+MemoryMax=800M
+MemorySwapMax=0
+Restart=always
+RestartSec=5
+EnvironmentFile=/etc/environment
+ExecStartPre=/bin/sh -c "systemctl set-environment HOSTNAME=$(hostname | tr [:upper:] [:lower:])"
```

Comment:
Dealing with the hostname will need to be done with some care, it is handled differently in the different infrastructures, some refs:
- https://github.com/gardener/gardener-extension-provider-openstack/blob/bbe65c6463add13a7f1bdbfd034e9970576ff2e5/pkg/webhook/controlplane/ensurer.go#L193
- https://github.com/gardener/gardener-extension-provider-gcp/blob/24467275e0bf2e6f8bc6713b5ba26339464094d0/pkg/webhook/controlplane/ensurer.go#L360
- https://github.com/gardener/gardener-extension-provider-aws/blob/5f900092991436b748ac8fc805adb3440d93ac54/pkg/webhook/controlplane/ensurer.go#L389

Reply:
Thanks for the proposal. It makes a lot of sense. Unifying the hostname is important in regard to showing the same observability signal field/attribute in logs. That is event more important to be unified with the observability logs from the kubelet service or any other node specific service that emits logs or metrics. Synergies are mandatory in this case. That being said we will check implementation details but the way kubelets fetch the hostnames shall be utilized also by the Collector as well, ensuring consistent field/attribute values. (That is not the case in current implementation with `valitail`)

Reply:
Agree. You can even remove the `ExecStartPre` line from the proposal.